Sources/Datadog/Traces/DDSpan.swift

```diff
@@ -9,14 +9,16 @@ import Foundation
 
 internal class DDSpan: Span {
     private(set) var operationName: String
-    /// The `Tracer` which created this span.
-    internal let issuingTracer: DDTracer
+    private(set) var tags: [String: Codable]
     internal let startTime: Date
     internal var isFinished: Bool = false
+    /// The `Tracer` which created this span.
+    private let issuingTracer: DDTracer
 
-    init(tracer: DDTracer, operationName: String, parentSpanContext: DDSpanContext?, startTime: Date) {
+    init(tracer: DDTracer, operationName: String, parentSpanContext: DDSpanContext?, tags: [String: Codable], startTime: Date) {
         self.issuingTracer = tracer
```

Comment:
if `Tracer` must be created by another `Tracer`, shouldn't its API be similar to the following?
```swift
private init(...) { ... } // no public/internal constructor
public func createTrace(...) -> Trace { ... } //public instance method to create a new Trace
```
the design above seems to enforce the correctness of `issuingTracer`

Reply:
I'm not sure if I understand, but the `DDSpan` cannot be created by the user. In fact, `DDSpan` is an `internal` type and users will only see `OpenTracing.Span`.

It is the `DDTracer` that creates `OpenTracing.Span` with `tracer.startSpan(...) -> OpenTracing.Span`.

Does it bring any answer 🤔?

Reply:
the problem with my proposal would be the following
```swift
// Span
fileprivate init(tracer, params...) { ... }
// Tracer extension
public issueSpan(params...) { ... }
```
we need to duplicate `params`. i saw this case as `Builder` pattern but it is not quite the same.

in short: let's go with the current code 👍💯